src/gt4py/next/ffront/decorator.py

```diff
@@ -472,8 +477,9 @@ def itir(self):
                 new_clos.inputs.pop(index)
             new_args = [ref(inp.id) for inp in new_clos.inputs]
             params = [sym(inp.id) for inp in new_clos.inputs]
-            for value in self.bound_args.values():
-                new_args.append(promote_to_const_iterator(literal_from_value(value)))
+            for key, value in self.bound_args.items():
+                pos = list(self.past_node.type.definition.pos_or_kw_args).index(key)
+                new_args.insert(pos, promote_to_const_iterator(literal_from_value(value)))
```

Comment:
looking at this I guess the answer to the comment above is: order matters?

Can't we reconstruct the order here?
When we go through the original closure.inputs, can't we just directly build the new_args with the correct order?